src/lib/macos-compat.c

```diff
@@ -0,0 +1,100 @@
+#include <fcntl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <unistd.h>
+#include <time.h>
+
+#include "macos-compat.h"
+
+// Time code based on: https://github.com/ChisholmKyle/PosixMachTiming/tree/master/src
+static inline void time_diff (const struct timespec* in, struct timespec* out) 
```

Comment:
i think we've already got this somewhere. there's definitely timespec_to_ns() and friends.

including this means we suddenly have ISC-licensed code, which (as you know) pokes at the sleeping FTPmaster dragon, especially until i'm a full DD. but it works if we need it, i guess.

Reply:
It's pretty generic code, I'm not sure if it means we need to mention ISC

Reply:
I went ahead and implemented `clock_nanosleep()` using `timespec_to_ns()` in the new `src/compat/compat.c`. It's only compiled on (non-Linux && non-FreeBSD), so it can be safely included in the source lists of all binaries (no CMake checks). If you merge, you ought just be able to kill this function. The others are fine, but let's move them to `src/compat` so as not to clutter up the CMakeLists.txt.